bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableVeneerSettingsFactory.java

```diff
@@ -0,0 +1,327 @@
+  private static CredentialsProvider buildCredentialProvider(
+      CredentialOptions credentialOptions) throws IOException {
+    try {
+      final Credentials credentials = CredentialFactory.getCredentials(credentialOptions);
+      if (credentials == null) {
+        LOG.info("Enabling the use of null credentials. This should not be used in production.");
+        return NoCredentialsProvider.create();
+      }
+
+      return FixedCredentialsProvider.create(credentials);
+    } catch (GeneralSecurityException exception) {
+      throw new IOException("Could not initialize credentials.", exception);
+    }
+  }
+
+  /** Builds {@link BatchingSettings} based on {@link BulkOptions} configuration. */
+  private static void buildBulkMutationsSettings(Builder builder, BigtableOptions options) {
+    BulkOptions bulkOptions = options.getBulkOptions();
+    BatchingSettings.Builder batchBuilder =
+        builder.bulkMutationsSettings().getBatchingSettings().toBuilder();
+
+    long autoFlushMs = bulkOptions.getAutoflushMs();
+    long bulkMaxRowKeyCount = bulkOptions.getBulkMaxRowKeyCount();
+    long maxInflightRpcs = bulkOptions.getMaxInflightRpcs();
+
+    if (autoFlushMs > 0) {
+      batchBuilder.setDelayThreshold(ofMillis(autoFlushMs));
+    }
+    FlowControlSettings.Builder flowControlBuilder = FlowControlSettings.newBuilder();
+    if (maxInflightRpcs > 0) {
+      flowControlBuilder
+          .setMaxOutstandingRequestBytes(bulkOptions.getMaxMemory())
+          .setMaxOutstandingElementCount(maxInflightRpcs * bulkMaxRowKeyCount);
+    }
+
+    batchBuilder
+        .setIsEnabled(bulkOptions.useBulkApi())
+        .setElementCountThreshold(Long.valueOf(bulkOptions.getBulkMaxRowKeyCount()))
+        .setRequestByteThreshold(bulkOptions.getBulkMaxRequestSize())
+        .setFlowControlSettings(flowControlBuilder.build());
+
+    RetrySettings retrySettings =
+        buildIdempotentRetrySettings(builder.bulkMutationsSettings().getRetrySettings(), options);
+
+    // TODO(rahulkql): implement bulkMutationThrottling & bulkMutationRpcTargetMs, once available
+    builder.bulkMutationsSettings()
+        .setBatchingSettings(batchBuilder.build())
+        .setRetrySettings(retrySettings)
+        .setRetryableCodes(buildRetryCodes(options.getRetryOptions()));
+  }
+
+  /**
+   * Builds BigtableDataSettings#checkAndMutateRowSettings when short timeout is other
+   * than 60_000 ms.
+   */
+  private static void buildCheckAndMutateRowSettings(Builder builder, long rpcTimeoutMs) {
+    if(rpcTimeoutMs != SHORT_TIMEOUT_MS_DEFAULT) {
+      builder.checkAndMutateRowSettings()
+          .setSimpleTimeoutNoRetries(ofMillis(rpcTimeoutMs));
+    }
+  }
+
+  /**
+   * Builds BigtableDataSettings#readModifyWriteRowSettings when short timeout is other
+   * than 60_000 ms.
+   */
+  private static void buildReadModifyWriteSettings(Builder builder, long rpcTimeoutMs) {
+    if(rpcTimeoutMs != SHORT_TIMEOUT_MS_DEFAULT) {
```

Comment:
why if? (here and above)

Reply:
If shortTimeOutMs is at default then assuming the client did not customize BigtableOption's settings, 
that is why we are not overriding the underlying settings here(or directly using defaults of GCJ's settings).

Reply:
I think you should override with the BigtableOptions in case the diverge

Reply:
> override with the BigtableOptions in case the diverge
By saying above comment,
do you mean we should check all configuration against their default value(present in BigtableOption) & then override only those which are different?
OR
Should we override only in case of specific operation(say non-streming operations) by checking BigtableOption's present value against BigtableDataSetting's default value?

Reply:
I just meant you should remove the if statement and always setSimpleTimeoutNoRetries disrarding if the if its the default value or not. If we decide that to change the gapic config to 30 secs, but forget to change SHORT_TIMEOUT_MS_DEFAULT we will have confusing behavior: the user will check their BigtableOptions and see SHORT_TIMEOUT_MS_DEFAULT = 1 minute, but their rpcs will exceed deadline @ 30secs

Reply:
Thanks for the clarification!! 
I followed the same [here](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/pull/1973/files#diff-6a492166d0005302ce3c542cdcf00ad5R96).